src/lavinmq/amqp/queue/queue.cr

```diff
@@ -744,6 +744,18 @@ module LavinMQ::AMQP
       end
     end
 
+    def unacked_messages
+      unacked_messages = consumers.each.select(AMQP::Consumer).flat_map do |c|
```

Comment:
They can, but we overload this in `MQTT::Session` and return an empty array, mainly because for `AMQP::Queue` a channel holds the `@unacked` messages. We don't have a channel for `MQTT::Session`, and we use different types for the `MQTT::Session` `@unacked`. 

Including `MQTT:Consumer` here here is going to require a bigger rewrite (which is planned for, and i know @spuun 
 has idéas with this PR  #873). 
Just don't think it should be blocking a release, so we overload and return empty for now.